src/react/provider.js

```diff
@@ -45,27 +45,29 @@ export class DiProvider extends Component {
     // supports global di if needed
     globalDi._fromProvider(use, { global });
     // support single or multiple targets
-    const targets = target && (Array.isArray(target) ? target : [target]);
+    const targets =
+      target && new WeakSet(Array.isArray(target) ? target : [target]);
 
     this.value = {
       getDependencies(realDeps, targetChild) {
         // First we collect dependencies from parent provider(s) (if any)
         const dependencies = getDependencies(realDeps, targetChild);
         // If no target or target is in the array of targets, map use
-        if (!targets || targets.includes(targetChild)) {
-          return dependencies.map((dep) => {
+        if (!targets || targets.has(targetChild)) {
```

Comment:
side note: this change really helps reducing "memory noise" letting specialised tools to provide clearer signal.
👍